<launch>
  <arg name="device_ip" default="192.168.1.200" />
  <arg name="msop_port" default="2368" />
  <arg name="difop_port" default="2369" />
  <arg name="time_synchronization" default="false" />

  <node pkg="lslidar_ch64w_driver" type="lslidar_ch64w_driver_node" name="lslidar_ch64w_driver_node" output="screen">
    <!-- param name="pcap" value="$(find lslidar_ch64w_driver)/pcap/xxx.pcap"/-->
    <param name="lidar_ip" value="$(arg device_ip)"/>
    <param name="msop_port" value="$(arg msop_port)" />
    <param name="difop_port" value="$(arg difop_port)"/>
    <param name="device_port" value="2368"/>
    <param name="add_multicast" value="false"/>
    <param name="group_ip" value="224.1.1.2"/>
    <param name="time_synchronization" value="$(arg time_synchronization)"/>
  </node>

  <node pkg="lslidar_ch64w_decoder" type="lslidar_ch64w_decoder_node" name="lslidar_ch64w_decoder_node" output="screen">
    <param name="frame_id" value="laser_link"/>
    <param name="point_num" value="2000"/>
    <param name="min_range" value="0.15"/>
    <param name="max_range" value="150.0"/>
    <param name="angle_disable_min" value="0"/>
    <param name="angle_disable_max" value="0"/>
    <Param name="pointcloud_topic" value="lslidar_point_cloud"/>
    <param name="frequency" value="10"/>
    <param name="pulse_repetition_rate" value="60.0"/>
    <param name="time_synchronization" value="$(arg time_synchronization)"/>
    <param name="channel_num" value="16"/>
    <param name="publish_point_cloud" value="true"/>
    <param name="publish_laserscan" value="false"/>
  </node>

  <!--node name="rviz" pkg="rviz" type="rviz" args="-d $(find lslidar_ch_decoder)/launch/lslidar_ch.rviz" output="screen"/-->

</launch>
